app/models/series.rb

```diff
@@ -867,16 +867,24 @@ def last_data_added_string
     last_data_added.strftime('%B %e, %Y')
   end
   
-  def Series.get_all_series_from_website(url_string)
-    series_from_website = (DataSource.get_all_uhero.where("eval LIKE '%#{url_string}%'").all.map {|ds| ds.series}).uniq
-    series_names = series_from_website.map {|s| s.name }
+  def Series.get_all_series_by_eval(patterns)
+    if patterns.class == String
+      patterns = [patterns]
+    end
+    series = []
+    patterns.each do |pat|
+      pat.gsub!('%', '\%')
+      series += DataSource.get_all_uhero.where("eval LIKE '%#{pat}%'").map{|ds| ds.series } rescue []
```

Comment:
Was this failing? It would be nice to know about failures here as they may be misspecified data sources that we want to load. Maybe add a logger.error to this.

Reply:
instead of mapping data sources to series, can we not use the activerecord relation?

Reply:
for example, `DataSource.where('eval LIKE "%bea.gov%"').joins(:series).pluck(:name).uniq` works for me.

Reply:
Points taken. Will modify.